src/pool/pool_jemalloc.c

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright (C) 2022-2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+
+#include <jemalloc/jemalloc.h>
+
+#include <umf/memory_pool.h>
+#include <umf/memory_pool_ops.h>
+#include <umf/pools/pool_jemalloc.h>
+
+#define MALLOCX_ARENA_MAX (MALLCTL_ARENAS_ALL - 1)
+
+#ifdef _WIN32
+#define __TLS __declspec(thread)
```

Comment:
btw, we define __TLS several times in our code - we could move this to "base.h" in the separate PR

Reply:
Done: https://github.com/oneapi-src/unified-memory-framework/pull/71